service/matching/task_queue_partition_manager.go

```diff
@@ -979,6 +984,7 @@ func (pm *taskQueuePartitionManagerImpl) getPhysicalQueuesForAdd(
 	forwardInfo *taskqueuespb.TaskForwardInfo,
 	runId string,
 	workflowId string,
+	isQuery bool,
```

Comment:
I think an enum for task type would be better (we probably have one), in case we would want to do something different for Activity, WF, Nexus, or Query task later on.

But also, it's not a breaking change to change this bool to an enum later, if a need ever arises, so if it's non-trivial to change this now, you should just keep it as is.